### PR TITLE
Don't use variations if DNT is set or tracking is disallowed

### PR DESCRIPTION
**Pull request by mandx**
## Description

This changes attempt to skip all the testing/variation selection code if we can detect that DoNotTrack is set in the browser or if the internal `window.builderNoTrack` flag is set. The idea is to provide a consistent user experience when tracking is disallowed/disabled, which in this case only the default variation should always be used.

### CODE REVIEW

File: packages/core/src/builder.class.ts [hunk 1]
```diff
@@ -535,7 +523,6 @@ export class Builder {
   static isBrowser = isBrowser;
   static isReactNative = isReactNative;
   static isServer = !isBrowser && !isReactNative;
-  static previewingModel = Builder.isBrowser && getQueryParam(location.href, 'builder.preview');
```
Comment (teleaziz) [inline packages/core/src/builder.class.ts:538]: why Is this change needed?

File: packages/core/src/builder.class.ts [hunk 2]
```diff
@@ -1513,10 +1500,8 @@ export class Builder {
     return this;
   }
 
-  get previewingModel() {
-    const search = this.getLocation().search;
-    const params = QueryString.parse((search || '').substr(1));
-    return params['builder.preview'];
+  get previewingModel(): string | undefined {
+    return Builder.isBrowser ? QueryString.parse((this.getLocation().search || '').substr(1))['builder.preview'] : undefined;
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:1504]: is this change necessary for this PR? I'm concerned with some of these changes being breaking and seemingly out of scope for this PR so could have slipped in silently and problematically

this change needs a revert, getLocation() supports server side _and_ client side, and those using server side rendering (e.g. next.js SSR, angular, etc) need the current behavior

Comment (steve8708) [reply packages/core/src/builder.class.ts:1504]: aka they expect a server side preview when applicable, or else they worry "why is the content flashing when I preview pages, I think I shouldn't publish this as there seems to be an issue"

**Comment by steve8708** [comment]
@mandx can we address the feedback and get this in

File: packages/react/src/components/variants-provider.component.tsx [hunk 3]
```diff
@@ -104,6 +104,8 @@ export const VariantsProvider: React.SFC<VariantsProviderProps> = ({
   initialContent,
   children,
 }) => {
+  if (!builder.canTrack) return children([initialContent]);
```
Comment (steve8708) [inline packages/react/src/components/variants-provider.component.tsx:107]: I don't know this area of the code so will need @teleaziz to chime in on

**Review by steve8708 — APPROVED** [review]

**Review by teleaziz — APPROVED** [review]
